src/appimagetool/appimagetool.go

```diff
@@ -362,32 +362,29 @@ func GenerateAppImage(appdir string) {
 	var iconfile string
 
 	// Check if we find a png matching the Icon= key in the top-level directory of the AppDir
+	// or at usr/share/icons/hicolor/256x256/apps/ in the AppDir
 	// We insist on a png because otherwise we need to costly convert it to png at integration time
 	// since thumbails need to be in png format
 	if helpers.CheckIfFileExists(appdir+"/"+iconname+".png") == true {
 		iconfile = appdir + "/" + iconname + ".png"
-	} else if helpers.CheckIfFileExists(appdir + "/usr/share/icons/hicolor/256x256/" + iconname + ".png") {
+	} else if helpers.CheckIfFileExists(appdir + "/usr/share/icons/hicolor/256x256/apps/" + iconname + ".png") {
 		// Search in usr/share/icons/hicolor/256x256 and copy from there
-		input, err := ioutil.ReadFile(appdir + "/usr/share/icons/hicolor/256x256/" + iconname + ".png")
+		input, err := ioutil.ReadFile(appdir + "/usr/share/icons/hicolor/256x256/apps/" + iconname + ".png")
```

Comment:
Why do you read the image here but not in the branch above? :thinking:

Reply:
Good question - probably just using `helpers.CheckIfFileExists` would be sufficient...

Reply:
That could also be outsourced to after the if. Just check whatever is in`fileName`. That way you'd also get a consistent behavior across all branches :)

Reply:
Care to send a PR @Krzmbrzl? I'd highly appreciate it.

Reply:
I can try - never worked with Go before ^^

Reply:
Looks like you succeeded :+1: Thank you.